apps/web/server/auth.ts

```diff
@@ -77,17 +88,34 @@ const providers: Provider[] = [
       email: { label: "Email", type: "email", placeholder: "Email" },
       password: { label: "Password", type: "password" },
     },
-    async authorize(credentials) {
+    async authorize(credentials, req) {
       if (!credentials) {
+        logAuthenticationError(
+          "<unknown>",
+          "Credentials missing",
+          req.headers?.["x-forwarded-for"],
+        );
         return null;
       }
 
       try {
-        return await validatePassword(
+        const isValidPassword = await validatePassword(
           credentials?.email,
           credentials?.password,
         );
+        logAuthenticationError(
+          credentials?.email,
+          "Password invalid",
+          req.headers?.["x-forwarded-for"],
+        );
```

Comment:
It's not clear to me why is this an authentication error? Also `validatePassword` returns the user or throws an error. So it's return value being `isValidPassword` is confusing.